### PR TITLE
sets up basic structure for creating a database

### PR DESCRIPTION
**Pull request by lollar**
There are still things to consider... like how to programatically decide
which database that you are using in the database connector. Also this
is still intertwined with postgres so I still need to figure out how to
unwrap that (eventually).

I'm not happy with how much is hardcoded in creating that database.  I
probably need to force ENV setup for database stuffs.

troubleshoots connection to newly created database

comments out the defaults method because y'know there is a bug :)

### CODE REVIEW

**Comment by lollar** [comment]
Finishes #2

File: src/lol-rm-migrations/create_database.cr [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require "yaml"
+require "db"
+require "pg"
+
+class CreateDatabase
+  def self.call(database_name)
+    new(database_name).run
+  end
+
+  DB_CONFIG_FILE = "/home/lollar/workspace/lol-rm-migrations/db/config.yml"
```
Comment (reecebisel) [inline src/lol-rm-migrations/create_database.cr]: This works for now, but we need to figure out a way to find this dynamically. Or pass in the file.

File: src/lol-rm-migrations/create_database.cr [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      db.close
+    end
+  end
+
+  def parsed_file
+    YAML.parse(File.read(DB_CONFIG_FILE))
+  end
+
+  def initial_database_connection
+    "postgres://#{parsed_file[@database_name]["host"]}:#{parsed_file[@database_name]["port"]}/postgres"
```
Comment (reecebisel) [inline src/lol-rm-migrations/create_database.cr]: I don't know if this changes from machine to machine or if its fairly regular to look like this.

File: src/lol-rm-migrations/create_database.cr (hunk 1 above)
Comment (lollar) [reply src/lol-rm-migrations/create_database.cr]: I think that requiring that the config is in a specified location or else it blows up should be standard for now and then we can look at other solutions eventually.  I'll get this to load dynamically before merge

File: src/lol-rm-migrations/create_database.cr (hunk 2 above)
Comment (lollar) [reply src/lol-rm-migrations/create_database.cr]: the things outside of the parsed file, i.e. `postgres://` and using the `postgres` user should be standard in postgres at least greater than 8 or 9. We'll probably need to look into it eventually. and if it is different then maybe we add a default block to the config file and also parse the values that are currently static.

**Pull request merged**